pkg/controllers/dsa/controller.go

```diff
@@ -225,6 +235,25 @@ func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 				},
 			},
 		})
+
+		if devicePlugin.Spec.ProvisioningConfig != "" {
```

Comment:
let's add checks in the validating webhook to warn users when user sets `ProvisioningConfig` but `InitImage` is empty